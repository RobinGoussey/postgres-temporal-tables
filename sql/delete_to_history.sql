CREATE EXTENSION IF NOT EXISTS psql_temporal;

CREATE TABLE delete_to_history_firemen_temporal (
	id serial PRIMARY KEY,
	name text,
	"rank" text
);
-- fill table
INSERT INTO delete_to_history_firemen_temporal (name,"rank") VALUES ('Bram','Sgt.');
-- create tables
select public.convert_to_temporal('delete_to_history_firemen_temporal');

-- should equal one: there is a current record in the history.
select count(id) from temporal_delete_to_history_firemen_temporal_history where sysendtime is null;

DELETE from delete_to_history_firemen_temporal where name = 'Bram';
-- should be 0, as now the last active record is closed.
select count(id) from temporal_delete_to_history_firemen_temporal_history where sysendtime is null;