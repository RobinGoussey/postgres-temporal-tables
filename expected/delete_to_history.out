CREATE EXTENSION IF NOT EXISTS psql_temporal;
NOTICE:  extension "psql_temporal" already exists, skipping
CREATE TABLE delete_firemen_temporal (
	id serial PRIMARY KEY,
	name text,
	"rank" text
);
-- fill table
INSERT INTO delete_firemen_temporal (name,"rank") VALUES ('Bram','Sgt.');
-- create tables
select public.convert_to_temporal('delete_firemen_temporal');
NOTICE:  trigger "delete_firemen_temporal_temporal_trigger" for relation "public.delete_firemen_temporal" does not exist, skipping
 convert_to_temporal 
---------------------
 t
(1 row)

-- should equal one: there is a current record in the history.
select count(id) from temporal_delete_firemen_temporal_history where sysendtime is null;
 count 
-------
     1
(1 row)

DELETE from delete_firemen_temporal where name = 'Bram';
-- should be 0, as now the last active record is closed.
select count(id) from temporal_delete_firemen_temporal_history where sysendtime is null;
 count 
-------
     0
(1 row)

